app/models/payment_gateway_setting.rb

```diff
@@ -37,4 +39,25 @@ def active_gateway_type
       errors.add(:gateway_type, :invalid)
     end
   end
+
+  def increment_failure_count
+    lock!
```

Comment:
This is a pessimistic lock we should try to avoid that unless we are sure that under spam conditions it does not end up with deadlocks or a hold query. This will for sure make the service down as the unicorn will all timeouts